torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -542,7 +543,19 @@ void ProcessGroupAgent::listenLoop() {
       recvWork_ = work;
     }
 
-    if (!rpcRunning_.load() || !work->wait() /* aborted */) {
+    if (!rpcRunning_.load()) {
+      return;
+    }
+    bool workCompleted;
+    try {
+      workCompleted = work->wait();
```

Comment:
Doing `rpcRunning_.store(false);` may not be the best approach here because this will mean that `::shutdown()` code will not run which joins threads etc. Is this addressing https://github.com/pytorch/pytorch/issues/29703? If so, it should probably be done in a separate PR. I was working on this issue as well so we should sync on how it can be best addressed.

Reply:
(See https://github.com/pytorch/pytorch/pull/32957)

Reply:
Can we back out the listenLoop changes from this PR and patch this PR on top of https://github.com/pytorch/pytorch/pull/32957? Its easier to review all the listenLoop changes in one place.